pycbc/filter/qtransform.py

```diff
@@ -0,0 +1,254 @@
+# Copyright (C) 2017  Hunter A. Gabbard
+# Most of this is a port of Duncan Macleod's GWPY qtransform.py script
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module retrives a timeseries and then calculates the q-transform of that time series
+
+Example
+-------
+    $ python q-transform.py -s 4096 -u test -o /Users/pycbc_qtransform
+
+"""
+
+from math import pi, ceil, log, exp
+import numpy as np
+from pycbc.types.timeseries import FrequencySeries, TimeSeries
+import os, sys
+from pycbc.frame import read_frame
+from pycbc.filter import highpass_fir, matched_filter
+from pycbc.waveform import get_fd_waveform
+from pycbc.psd import welch, interpolate
+from pycbc.fft import ifft
+import urllib
+import datetime
+from scipy.interpolate import (interp2d, InterpolatedUnivariateSpline)
+from numpy import fft as npfft
+import argparse
+import datetime
+
+from matplotlib import use
```

Comment:
Done. Though some of the imports are used in the larger functions I haven't committed yet.